From: Mel Gorman <mgorman@suse.de>
Date: Tue, 2 Oct 2018 13:49:33 +0100
Subject: [PATCH] sched/numa: Stop multiple tasks from moving to the CPU at the
 same time kabi

References: bnc#1101669 optimise numa balancing for fast migrate
Patch-mainline: No, queued in subsystem maintainer tree

This should be safe as no module should ever be accessing struct rq or
have a copy of it.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 kernel/sched/sched.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 99276d0ee2d8..0ccac55fb5bf 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -647,7 +647,9 @@ struct rq {
 #ifdef CONFIG_NUMA_BALANCING
 	unsigned int nr_numa_running;
 	unsigned int nr_preferred_running;
+#ifndef __GENKSYMS__
 	unsigned int numa_migrate_on;
+#endif
 #endif
 	#define CPU_LOAD_IDX_MAX 5
 	unsigned long cpu_load[CPU_LOAD_IDX_MAX];
